analyzer/codechecker_analyzer/pre_analysis_manager.py

```diff
@@ -114,8 +111,22 @@ def pre_analyze(params):
                 ctu_triple_arch.get_triple_arch(action, action.source,
                                                 clangsa_config,
                                                 analyzer_environment)
-            ctu_manager.generate_ast(triple_arch, action, action.source,
-                                     clangsa_config, analyzer_environment)
+
+            # TODO: reorganize the various ctu modes parameters
+            # Dump-based analysis requires serialized ASTs.
+            if clangsa_config.ctu_on_demand:
+                ctu_manager.generate_invocation_list(triple_arch, action,
```

Comment:
is the invocation list different from the file generated by the extdef-mapping tool?

Reply:
Yes, the invocation list file is a mapping from the source file to the clang invocation, the extdef-mapping file is a mapping from usr-names to source files (or ast dumps in case of dump-based analysis).